### ISSUE TITLE
MySqlBulkLoader inside Transaction throws InvalidOperationException

### ISSUE THREAD

**Issue by alextruhan** [mysql-net/MySqlConnector#300]
Invokation of MySqlBulkLoader.Load() inside Transaction throws InvalidOperationException with message "The transaction associated with this command is not the connection's active transaction.".

```csharp
m_database.Connection.Open();
var tr = m_database.Connection.BeginTransaction();
var bl = new MySqlBulkLoader(m_database.Connection)
{
     SourceStream = new MemoryStream(m_memoryStreamBytes),
     TableName = m_testTable,
};
bl.Load();//InvalidOperationException is throwed here
tr.Rollback();
```

Due to code from MySqlBulkLoader.LoadAsync
```csharp
                var commandString = BuildSqlCommand();
                var cmd = new MySqlCommand(commandString, Connection)
                {
                    CommandTimeout = Timeout
                };
```
And code from MySqlCommand.IsValid
```csharp
		public MySqlCommand(string commandText, MySqlConnection connection, MySqlTransaction transaction)
		{
			CommandText = commandText;
			DbConnection = connection;
			DbTransaction = transaction;
			m_parameterCollection = new MySqlParameterCollection();
			CommandType = CommandType.Text;
		}
```
And
```csharp
else if (DbTransaction != Connection.CurrentTransaction)
				exception = new InvalidOperationException("The transaction associated with this command is not the connection's active transaction.");
```

This exception will be throw always when MySqlBulkLoader.Load will be invoked inside transaction. 
There are some workarounds to fix this issue?
Or do I do something wrong?

**Comment by bgrainger** [comment]
I thought I had read in the MySQL manual that `LOAD DATA` wasn't supported inside a transaction, but I can't find any such restriction now. This sounds like it might be a bug; I'll have to check the actual behaviour of `LOAD DATA` with `COMMIT` and `ROLLBACK`.

**Comment by alextruhan** [comment]
But MySqlBulkLoader from package MySql.Data work well inside transaction without any excepcions.

**Comment by bgrainger** [comment]
Fixed in [0.24.0](https://github.com/mysql-net/MySqlConnector/releases/tag/0.24.0).

**Comment by GomesNayagam** [comment]
@bgrainger @alextruhan  looks the issue again opened in 0.31.0, check again.
System.InvalidOperationException: The transaction associated with this command is not the connection's active transaction.

**Comment by bgrainger** [comment]
I suspect this is a regression caused by fixing #389.

**Comment by bgrainger** [comment]
Should be fixed in [0.31.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.31.1).

If not, please open a new issue with sample code to reproduce the problem.
